crates/biome_service/src/file_handlers/svelte.rs

```diff
@@ -21,22 +23,36 @@ use super::{SearchCapabilities, parse_lang_from_script_opening_tag};
 #[derive(Debug, Default, PartialEq, Eq)]
 pub struct SvelteFileHandler;
 
-// https://regex101.com/r/E4n4hh/6
+// Kept for backward compatibility with existing code
 pub static SVELTE_FENCE: LazyLock<Regex> = LazyLock::new(|| {
-    Regex::new(r#"(?ixs)(?<opening><script(?:\s.*?)?>)\r?\n?(?<script>(?U:.*))</script>"#).unwrap()
+    Regex::new(r#"(?ixs)(?<opening><script(?:\s.*?)?>)\r?\n?(?<script>(?U:.*?))</script>"#).unwrap()
 });
 
 impl SvelteFileHandler {
-    /// It extracts the JavaScript/TypeScript code contained in the script block of a Svelte file
+    /// Extracts the JavaScript/TypeScript code contained in the script block of a Svelte file
+    /// using HTML parsing instead of regex for better handling of HTML structure.
     ///
     /// If there's no script block, an empty string is returned.
     pub fn input(text: &str) -> &str {
+        // For backwards compatibility, still use the regex approach
         match Self::matches_script(text) {
             Some(script) => &text[script.start()..script.end()],
             _ => "",
         }
     }
 
+    /// Extracts all embedded JavaScript content from a Svelte file using HTML parsing.
+    /// This provides better handling of HTML structure and script indentation.
+    pub fn extract_scripts(text: &str, cache: &mut NodeCache) -> Vec<EmbeddedJsContent> {
+        // Parse the Svelte file as HTML
+        let html_file_source = HtmlFileSource::default();
+        let parse = parse_html_with_cache(text, html_file_source, cache);
+        let html_root = HtmlRoot::cast(parse.syntax()).expect("Failed to cast to HtmlRoot");
+        
+        // Extract embedded scripts using the HTML handler
+        html::extract_embedded_scripts(&html_root, cache)
+    }
```

Comment:
This provides less functionality than the regex solution. The regex solution is able to understand if the internal code must be parsed as TypeScript or JavaScript. This solution doesn't do that.